### PR TITLE
[SPARK-27974][SQL] Support ANSI Aggregate Function: array_agg

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR used to support ANSI aggregate Function: `array_agg`.

The mainstream database supports `array_agg` show below:
**Presto**
https://prestodb.io/docs/current/functions/aggregate.html
**Teradata**
https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/Xffi_w_BZeSoaXg6Fd76Qw
**Snowflake**
https://docs.snowflake.com/en/sql-reference/functions/array_agg.html
**Oracle**
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/JSON_ARRAYAGG.html#GUID-6D56077D-78DE-4CC0-9498-225DDC42E054
**H2**
http://www.h2database.com/html/functions-aggregate.html#array_agg
**Postgresql**
https://www.postgresql.org/docs/8.4/functions-aggregate.html
**Maria**
https://mariadb.com/kb/en/json_arrayagg/
**Intersystems**
https://docs.intersystems.com/latest/csp/docbook/DocBook.UI.Page.cls?KEY=RSQL_jsonarrayagg
**MySQL**
https://www.mysqltutorial.org/mysql-aggregate-functions.aspx
**Druid**
https://druid.apache.org/docs/latest/querying/sql.html#aggregation-functions

### Why are the changes needed?
`array_agg` is very useful.

### Does this PR introduce _any_ user-facing change?
'Yes'. New feature.

### How was this patch tested?
New tests.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @MaxGekk @gengliangwang  cc @cloud-fan

File: sql/core/src/test/resources/sql-tests/inputs/group-by.sql [hunk 1]
```diff
@@ -204,3 +204,24 @@ FROM VALUES
...
+  collect_list(col),
+  array_agg(col)
+FROM VALUES
+  (1), (2), (1) AS tab(col);
+SELECT
+  a,
+  collect_list(b),
+  array_agg(b)
+FROM VALUES
+  (1,4),(2,3),(3,1),(4,2) AS v(a,b)
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/group-by.sql:219]: I would write only one test with GROUP BY, which has duplicated values
```
(1,4),(2,3),(1,4),(2,4)
```

Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/group-by.sql:219]: OK

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@cloud-fan Thank you very much.

**Comment by cloud-fan** [comment]
> Does this PR introduce any user-facing change?
> 'No'. New feature.

Please be more careful when writing PR descriptions. How can this be not user-facing?

**Pull request closed without merging**